File: WDF-Server/src/main/java/com/nohorang/suryongfootprint/model/Count.java
package com.nohorang.suryongfootprint.model;

import lombok.Getter;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;

@Getter
@Setter
@Entity
@Table(name = "COUNT")
public class Count {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int count_id;

    @ManyToOne
    @JoinColumn(name="user_id", nullable=false)
    private User user;

    @ManyToOne
    @JoinColumn(name="challenge_id", nullable=false)
    private Challenge challenge;

    @Column(name="challenge_count", nullable=false)
    private int challengeCount;

    @Column(name="post_count",nullable=false)
    private int postCount;

    @Column(name="approval_count", nullable=false)
    private int approvalCount;
}

File: WDF-Android/app/src/main/java/kr/co/nohorang/suryongfootprint/data/Count.kt
package kr.co.nohorang.suryongfootprint.data

import com.google.gson.annotations.SerializedName
import java.io.Serializable

// JSON 타입 변환에 사용될 객체(DTO)
// request의 편의성을 고려해서 선언할 것.
data class Count (@SerializedName("CountId")var count_id: String?,
                 @SerializedName("userId")var user_id: String?,
                 @SerializedName("challengeId")var challenge_id: Int?,
                 @SerializedName("challengeCount")var challenge_count: Int?,
                 @SerializedName("postCount")var post_count:Int?,
                  @SerializedName("approvalCount")var approval_count:Int?
) :Serializable {
}
